source/fem-dem/gls_sharp_navier_stokes.cc

```diff
@@ -3282,11 +3282,20 @@ GLSSharpNavierStokesSolver<dim>::sharp_edge()
                           // used for the definition of the stencil
                           // ("stencil_cell") is on a face between the cell_cut
                           // that is cut ("cell_cut") and the "stencil_cell".
-                          bool point_in_cell = cell_cut->point_inside(
-                            interpolation_points
-                              [stencil.number_of_interpolation_support_points(
-                                 order) -
-                               1]);
+                          bool point_in_cell;
+                          // The extrapolation can be disabled for debugging
+                          // purposes, although in most cases it shouldn't since
+                          // it is a core part of this solver
+                          if (this->simulation_parameters.particlesParameters
+                                ->enable_extrapolation)
+                            point_in_cell = cell_cut->point_inside(
+                              interpolation_points
+                                [stencil.number_of_interpolation_support_points(
+                                   order) -
+                                 1]);
+                          else
+                            point_in_cell = cell_cut->point_inside(
+                              support_points[local_dof_indices[i]]);
```

Comment:
I am not a fan of having parameters creep very deeply into loops like that, but ok...

Reply:
Me neither
If I think of a better way I'll change it in a future PR